pytest-target/target/azure.py

```diff
@@ -0,0 +1,159 @@
+"""Provides an `Azure(Target)` implementation using the Azure CLI."""
```

Comment:
I was thinking along the same lines. We can have an `AzureCLI` class and an `AzureAPI` class. That keeps their dependencies and implementation separate and decoupled. The CLI is the easiest for users, as it can just run using their existing CLI setup, and the API is the "advanced" usage scenario.